scripts/build.sh

```diff
@@ -1,12 +1,58 @@
 #!/bin/bash
+# Single source of truth for building Firecloud-Ui.
+# @ Jackie Roberti and Isaac Zarsky
+#
+# Provide command line options to do one or several things:
+#   compile: compile clojure proj into /target
+#   -d | --docker : provide arg either "build" or "push", to build and push docker image
+# Jenkins build job should run with all options, for example,
+#   ./docker/build.sh compile -d push
+
+
 IFS=$'\n\t'
 set -euxo pipefail
 
-docker run --rm \
-  -w /work \
-  -v "$PWD":/work \
-  -v jar-cache:/root/.m2 \
-  -v node-modules:/work/node_modules \
-  -e npm_config_unsafe_perm="true" \
-  broadinstitute/clojure-node \
-  bash -c 'lein with-profile deploy do clean, resource, cljsbuild once && npm install && npm run webpack -- -p'
+function clj_build() {
+    docker run --rm \
+      -w /work \
+      -v "$PWD":/work \
+      -v maven-cache:/root/.m2 \
+      -e npm_config_unsafe_perm="true" \
+      broadinstitute/clojure-node \
+      lein with-profile deploy do clean, resource, npm install, cljsbuild once
```

Comment:
As it was previously:
`bash -c 'lein with-profile deploy do clean, resource, cljsbuild once && npm install && npm run webpack -- -p'`